backend-zs/controllers/projects.controller.js

```diff
@@ -34,6 +37,75 @@ exports.createProject = function (req, res) {
     }).catch(error => res.status(500).json({ error: error }))
 }
 
+exports.submitParticipation = function (req, res) {
+    req.body.participant.confirmationToken = crypto.randomBytes(24).toString('hex');
+    const activationLink = process.env.ZS_URL + '/participation/verify/' + req.params.projectId + '/' + req.body.participant.confirmationToken;
+
+    Project.findOneAndUpdate({ _id: req.params.projectId }, { $push: { participants: req.body.participant } }, { useFindAndModify: false })
+        .then(project => {
+            const request = mailjet.post('send', { version: 'v3.1' }).request({
+                "Messages": [{
+                    "From": {
+                        "Email": "felix@zukunftschreiben.org",
+                    },
+                    "To": [{
+                        "Email": req.body.userEmail,
+                    }],
+                    "Subject": `Teilnahme am Project ${project.info.title} bestätigen`,
+                    "HTMLPart": `Vielen Dank für dein Interesse an der Teilnahme am Projekt ${project.info.title}. Klicke den <a href=${activationLink}>Link</a> und bestätige deine Teilnahme!`
+                }]
+            })
+
+            request.then(_ => { res.json('participant submission send successfully'); })
+                .catch(error => res.status(500).json({ error: error.message }))
+        })
+        .catch(error => res.status(500).json({ error: error.message }))
+}
+
+exports.verifyParticipation = async function (req, res) {
```

Comment:
Can we use a different statusCode for the unsuccessful responses

Reply:
Changed to **501** for `findOneAndUpdate` and **503** for `request`.

**501** Not Implemented
The server either does not recognize the request method, or it lacks the ability to fulfil the request.

**503** Service Unavailable
The server cannot handle the request.

Reply:
I was referring to the following responses:

`return res.json({ success: false, message: 'Project not found' })`  at lines 71, 82, 87, 106. Sorry if I wasn't clear about it.

Reply:
@klarmannsonia did you check my last reply here. I am afraid you resolved it before I posted my reply.